src/asm/lexer.c

```diff
@@ -219,6 +216,30 @@ YY_BUFFER_STATE yy_create_buffer(FILE *f)
 		}
 	}
 
+	pBuffer->nBufferSize = size;
+
+	/* Add newline if file doesn't end with one */
+	if (size == 0 || pBuffer->pBuffer[size - 1] != '\n') {
+		pBuffer->pBuffer[pBuffer->nBufferSize] = '\n';
+		pBuffer->nBufferSize++;
+	}
+
+	/* Add newline if \ will eat the last newline */
+	if (pBuffer->nBufferSize >= 2) {
+		size_t pos = pBuffer->nBufferSize - 2;
+
+		/* Skip spaces */
+		while (pos > 0 && pBuffer->pBuffer[pos] == ' ')
+			pos--;
+
+		if (pBuffer->pBuffer[pos] == '\\') {
+			pBuffer->pBuffer[pBuffer->nBufferSize] = '\n';
```

Comment:
So you allocate ``pBuffer->nBufferSize`` bytes, then write to ``pBuffer->pBuffer[pBuffer->nBufferSize]`` and "increase" the size by incrementing ``pBuffer->nBufferSize``, and then write 0 to the new end, which isn't allocated. This is wrong.

Reply:
I've changed the code a bit to make what it more obvious what's happening, but the behavior has not been changed.